Decompile to Text Editor.py

```diff
@@ -1,20 +1,54 @@
 #!/usr/bin/python3
 
+import json
 import subprocess
 import tempfile
+from pathlib import Path
 
 
 # ---------------------------------------------------------------------------
+# Configuration
 
-EDITOR_CLI = {
-    'BBEdit': ['/usr/local/bin/bbedit'],
-    'MacVim': ['/Applications/MacVim.app/Contents/MacOS/Vim', '-g'],
-    'SubEthaEdit': ['/usr/local/bin/see'],
-    'Sublime Text': ['/usr/local/bin/subl'],
-    'TextMate': ['/usr/local/bin/mate'],
-    'Visual Studio Code': ['/usr/local/bin/code']
+DEFAULT_CONFIG = {
+    'editor': 'Sublime Text',
+    'editors': {
+        'BBEdit': ['/usr/local/bin/bbedit'],
+        'MacVim': ['/Applications/MacVim.app/Contents/MacOS/Vim', '-g'],
+        'SubEthaEdit': ['/usr/local/bin/see'],
+        'Sublime Text': ['/usr/local/bin/subl'],
+        'TextMate': ['/usr/local/bin/mate'],
+        'Visual Studio Code': ['/usr/local/bin/code']
+    }
 }
-EDITOR = EDITOR_CLI['Sublime Text']
+
+
+def get_config_path():
+    script = Path(__file__)
+    filename = script.stem + '.json'
+    filepath = script.parent.joinpath(filename)
+    return filepath
+
+
+def read_config():
+    config_path = get_config_path()
+    if not config_path.is_file():
+        save_config(DEFAULT_CONFIG)
+    return json.loads(config_path.read_text())
+
+
+def save_config(data):
+    config_path = get_config_path()
+    _ = config_path.write_text(json.dumps(data, indent=2))
+
+
+def get_editor_command():
+    try:
+        config = read_config()
+    except:
+        config = DEFAULT_CONFIG
```

Comment:
There are plenty reasons a configuration file could not be created or JSON data could be invalid. This is some cheap user-proof / system-proof failguard.